crates/factor-wasi/src/lib.rs

```diff
@@ -190,11 +191,8 @@ impl InstanceBuilder {
     }
 
     /// Sets the WASI `stdout` descriptor to the given [`AsyncWrite`]r.
-    pub fn stdout_pipe(&mut self, w: impl AsyncWrite + Send + Unpin + 'static) {
-        self.stdout(AsyncStdoutStream::new(AsyncWriteStream::new(
-            1024 * 1024,
-            w,
-        )));
+    pub fn stdout_pipe(&mut self, w: impl Write + Send + Sync + Unpin + 'static) {
```

Comment:
Maybe rename this to `stdout_sync`? :shrug:

Reply:
Does `pipe` refer to the asyncness? I thought `pipe` fits just as much as it did before. I don't have strong feelings though either way.

Reply:
I don't know. It's all pipes in the end isn't it?